### PR TITLE
coerce estimator_label to character in reshape_diagnosis

### LINKED ISSUE #395: coerce estimator_label to factor?
Factors are creeping into diagnosis and causing problems including in `reshape_diagnosis` via `estimator_label` in custom estimators (and custom estimands), e.g.:

```
population <- declare_population(N = 100)

estimator <- declare_estimator(handler = function(data) {
  data.frame(estimator_label = "my_estimator", est = 5)
})

simulate_design(population + estimator, sims = 5) %>% as_tibble %>% head

# A tibble: 5 x 4
  design_label sim_ID estimator_label   est
  <chr>         <int> <fct>           <dbl>
1 design_1          1 my_estimator        5
2 design_1          2 my_estimator        5
3 design_1          3 my_estimator        5
4 design_1          4 my_estimator        5
5 design_1          5 my_estimator        5
```

Suggest we coerce with `as.character` at the `get_estimates/draw_estimates` stage. Objections?

Of course this can be fixed by the user via `stringsAsFactors = FALSE` but seems people forget this often.

### PR DESCRIPTION
**Pull request by graemeblair**
closes https://github.com/DeclareDesign/DeclareDesign/issues/395

### CODE REVIEW

File: R/get_functions.R [hunk 1]
```diff
@@ -83,7 +83,20 @@ get_function_internal <- function(design, data = NULL, start, end, pred, results
   
   design_subset <- Filter(pred, design[start:end])
   
-  run_design_internal.design(design_subset, current_df = data, results = results)[[what]]
+  ret <- run_design_internal.design(design_subset, current_df = data, results = results)[[what]]
   
-}
+  if(what == "estimates_df" && class(ret$estimator_label) != "character"){
```
Comment (nfultz) [inline R/get_functions.R]: Labels are optional, so don't print the warning if null.

Comment (graemeblair) [reply R/get_functions.R]: ah, thanks

Comment (nfultz) [reply R/get_functions.R]: Also character isn't really a class, its either a type or mode, I forget which.

Comment (graemeblair) [reply R/get_functions.R]: ah yes thanks will fix

**Pull request merged**